scout/server/blueprints/variants/controllers.py

```diff
@@ -229,6 +232,97 @@ def _get_group_assessments(store, case_obj, variant_obj):
     return group_assessments
 
 
+def render_variants_page(
+    category,
+    institute_id,
+    case_name,
+    form_builder,
+) -> dict:
+    """Shared logic for SV and MEI variants routes."""
+    page = get_variants_page(request.form)
+    institute_obj, case_obj = institute_and_case(store, institute_id, case_name)
+
+    # Variant type handling
+    variant_type = Markup.escape(
+        request.args.get("variant_type", request.form.get("variant_type", "clinical"))
+    )
+    if variant_type not in ["clinical", "research"]:
+        variant_type = "clinical"
+
+    variants_stats = store.case_variants_count(case_obj["_id"], institute_id, variant_type, False)
+
+    set_hpo_clinical_filter(case_obj, request.form)
+
+    if "dismiss_submit" in request.form:
+        dismiss_variant_list(
+            store,
+            institute_obj,
+            case_obj,
+            "variant.sv_variant",  # both SV and MEI use this
+            request.form.getlist("dismiss"),
+            request.form.getlist("dismiss_choices"),
+        )
+
+    activate_case(store, institute_obj, case_obj, current_user)
+
+    # Build form (different per category)
+    form = form_builder(store, institute_obj, case_obj, category, variant_type)
+
+    # Populate common filter choices
+    populate_chrom_choices(form, case_obj)
+    populate_institute_soft_filters(form=form, institute_obj=institute_obj)
+
+    genome_build = get_case_genome_build(case_obj)
+    cytobands = store.cytoband_by_chrom(genome_build)
+
+    update_form_hgnc_symbols(store, case_obj, form)
+
+    variants_query = store.variants(
+        case_obj["_id"], category=category, query=form.data, build=genome_build
+    )
+    result_size = store.count_variants(
+        case_obj["_id"], form.data, None, category, build=genome_build
+    )
+
+    if request.form.get("export"):
+        return download_variants(store, case_obj, variants_query)
+
+    data = sv_mei_variants(
+        store=store,
+        institute_obj=institute_obj,
+        case_obj=case_obj,
+        variants_query=variants_query,
+        page=page,
+    )
+
+    dismiss_variant_options = (
+        {**DISMISS_VARIANT_OPTIONS, **CANCER_SPECIFIC_VARIANT_DISMISS_OPTIONS}
+        if category == "cancer_sv"
+        else DISMISS_VARIANT_OPTIONS
+    )
```

Comment:
This list is populated correctly before now

Reply:
Right, this should work. But, when at some point we have many conditionals on category we must refactor. 😊